components/private/State/src/StateScheme.swift

```diff
@@ -0,0 +1,407 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+enum StateTheme: String {
+
+  case onWhite, onPrimary
+  case outline, elevated
+
+  // Some themes "inherit" attributes from other themes (only 1 level
+  // of inheritance is supported at this point):
+  var baseTheme: StateTheme? {
+    switch self {
+    case .outline: return .onWhite
+    case .elevated: return .onWhite
+    default: return nil
+    }
+  }
+}
+
+// State dictionary is a list of attributes. An attribute is a list of resources:
+typealias ResourceDictionary = [String:String]
```

Comment:
I worry about the robustness of a `typealiased` dictionary, could it be something else?

Reply:
It was easier to keep it a String in this implementation, but for the actual implementation, I agree this might be too limiting. I will add it as a todo item for the Alpha implementation.